### PR TITLE
Refactoring action forward logic

### PR DESCRIPTION
**Pull request by ncounter**
Depends on https://github.com/openSUSE/open-build-service/pull/15794#

Refactoring action forward logic

### CODE REVIEW

**Review by eduardoj — APPROVED** [review]

File: src/api/app/models/bs_request_action_submit.rb [hunk 1]
```diff
@@ -149,39 +149,21 @@ def creator_is_target_maintainer
...
-          end
-        end
-        unless link_is_already_devel
-          forward_object ||= []
-          forward_object << { project: linkinfo['project'], package: linkinfo['package'], type: 'link' }
-        end
-      end
+    linkinfo = target_package_object.linkinfo
+    # add all the devel packages into the forwards
+    forward_object = target_package_object.developed_packages.map do |dev_pkg|
```
Comment (danidoni) [inline src/api/app/models/bs_request_action_submit.rb:157]: I think `map` will return `[]` so the `forward_object || []` is reduntant. I haven't tested it though

Comment (ncounter) [reply src/api/app/models/bs_request_action_submit.rb:157]: I agree, thanks for the catch. It's a leftover after the IDE auto-fix.

File: src/api/app/models/bs_request_action_submit.rb [hunk 2]
```diff
@@ -149,39 +149,21 @@ def creator_is_target_maintainer
...
+      { project: dev_pkg.project.name, package: dev_pkg.name, type: 'devel' }
+    end
+    # check if the link is already in the forwards, add it otherwise
+    if linkinfo && (forward_object.none? { |forward| forward[:project] == linkinfo['project'] && forward[:package] == linkinfo['package'] })
```
Comment (danidoni) [inline src/api/app/models/bs_request_action_submit.rb:161]: So, we could do this, right?

```suggestion
    return forward_object unless linkinfo
    
    if forward_object.none? { |forward| forward[:project] == linkinfo['project'] && forward[:package] == linkinfo['package'] }
```

Comment (ncounter) [reply src/api/app/models/bs_request_action_submit.rb:161]: We could, sure, but why splitting a condition logic into two pieces?

Comment (hennevogel) [reply src/api/app/models/bs_request_action_submit.rb:161]: because return early makes code more readable for humans.

Comment (danidoni) [reply src/api/app/models/bs_request_action_submit.rb]: The main idea here is that when you are reading code, processing the first return you are removing cognitive load from the second condition. Whatever we do that removes cognitive load from code, the better.

**Review by danidoni — APPROVED** [review]

**Pull request merged**